### ISSUE TITLE
Daily master CLI fails to build CoreCLR

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/cli#8699]
## Steps to reproduce

```
cd git/coreclr
git checkout 2f14b350960

git clean -x -d -f

#update DotnetCLIVersion.txt to latest daily tag (2.1.300-preview2-008265)
vi DotnetCLIVersion.txt

./build.sh -verbose
```

## Expected  behavior

Successful build (works with 2.0.0)

## Actual behavior
```
     4>Target "CDefineChecker" in project "/home/vmjenkins/workspace/Dotnet/build-bootstrap/src/mscorlib/System.Private.CoreLib.csproj" (target "Build" depends on it):
       Using "Exec" task from assembly "Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
       Task "Exec"
         python /home/vmjenkins/workspace/Dotnet/build-bootstrap/src/mscorlib/../scripts/check-definitions.py "/home/vmjenkins/workspace/Dotnet/build-bootstrap/bin/obj/Linux.x64.Debug/System.Private.CoreLib/../cmake.definitions" "CODE_ANALYSIS;_LOGGING;DEBUG;TRACE;BIT64;
AMD64;;CORECLR;_USE_NLS_PLUS_TABLE;RESOURCE_SATELLITE_CONFIG;CODE_ANALYSIS_BASELINE;FEATURE_ARRAYSTUB_AS_IL;FEATURE_MULTICASTSTUB_AS_IL;FEATURE_STUBS_AS_IL;FEATURE_COREFX_GLOBALIZATION;FEATURE_MANAGED_ETW;FEATURE_MANAGED_ETW_CHANNELS;FEATURE_PAL;FEATURE_PERFTRACING;FEATU
RE_EVENTSOURCE_XPLAT;FEATURE_DEFAULT_INTERFACES;PROFILING_SUPPORTED;PLATFORM_UNIX;SIGNED" "" 
         /bin/sh: 1: export LANG=en_US.UTF-8; export LC_ALL=en_US.UTF-8; . /tmp/tmp1d57e5aa1130430c8d82cbf17024c398.exec.cmd: not found
```

Running the python command manually succeeds.

It looks like the master daily CLI build is failing to capture command output.

There are also 4 similar errors earlier in the log when creating version.cpp.  Looks like the shell scripts to gather environment data are failing with similar issue.

```
     1>Target "CreateOrUpdateCurrentVersionFile" in file "/home/vmjenkins/workspace/Dotnet/build-bootstrap/Tools/versioning.targets" from project "/home/vmjenkins/workspace/Dotnet/build-bootstrap/build.proj" (target "CreateVersionFileDuringBuild" depends on it):
       Using "Exec" task from assembly "Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
       Task "Exec"
         git rev-parse HEAD 2>&1
         /bin/sh: 1: export LANG=en_US.UTF-8; export LC_ALL=en_US.UTF-8; . /tmp/tmpe361845db42941bc8b4901500cc275cf.exec.cmd: not found
         The command "git rev-parse HEAD 2>&1" exited with code 127.
       Done executing task "Exec".
       Using "MakeDir" task from assembly "Microsoft.Build.Tasks.Core, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a".
       Task "MakeDir"
       Done executing task "MakeDir".
       Task "WriteLinesToFile" skipped, due to false condition; ('$(UseSourceLink)' == 'true' AND '$(GitHubRepositoryUrl)' != '' AND '$(LatestCommit)' != 'N/A' AND !Exists('$(SourceLinkFilePath)')) was evaluated as ('true' == 'true' AND 'https://github.com/dotnet/coreclr' != '' AND 'N/A' != 'N/A' AND !Exists('/home/vmjenkins/workspace/Dotnet/build-bootstrap/bin/obj/source_link.json')).
       Task "Exec"
         git show -s --format=%cd --date=short HEAD
         /bin/sh: 1: export LANG=en_US.UTF-8; export LC_ALL=en_US.UTF-8; . /tmp/tmp66b52bec08bb48c4afda033c58f5a44d.exec.cmd: not found
         The command "git show -s --format=%cd --date=short HEAD" exited with code 127.
       Done executing task "Exec".
       Task "Exec"
         whoami
         /bin/sh: 1: export LANG=en_US.UTF-8; export LC_ALL=en_US.UTF-8; . /tmp/tmpd872373412d04bd999576961d80aeb2d.exec.cmd: not found
         The command "whoami" exited with code 127.
       Done executing task "Exec".
       Task "Exec"
         hostname
         /bin/sh: 1: export LANG=en_US.UTF-8; export LC_ALL=en_US.UTF-8; . /tmp/tmpeca59cb50ec148059cdbcc75cfe0eba6.exec.cmd: not found
         The command "hostname" exited with code 127.
```

## Environment data

Linux Ubuntu 16.04 x64

coreclr hash 2f14b350960 

DotnetCLIVersion.txt 2.1.300-preview2-008265 (also tried 2.1.300-preview2-008263)

@jkotas If I understood https://github.com/dotnet/coreclr/issues/16299 correctly, you indicated this should work.

I am not sure where this issue belongs.  cli? , MSBuild? core-setup? Coreclr? soure-build?
Microsoft.Build.Tasks.Core ?

**Comment by sdmaclea** [comment]
@joperezr 
This may be fixed by https://github.com/dotnet/coreclr/pull/16101 

Checking

**Comment by sdmaclea** [comment]
Resolved by dotnet/coreclr#16101
